src/proposal.mligo

```diff
@@ -237,8 +236,8 @@ let unfreeze_proposer_and_voter_token
   Map.fold do_unfreeze proposal.voters store
 
 [@inline]
-let is_time_reached (proposal, sec : proposal * seconds): bool =
-  Tezos.now > proposal.start_date + int(sec)
+let is_level_reached (proposal, target : proposal * blocks): bool =
+  Tezos.level > proposal.start_level.blocks + target.blocks
```

Comment:
@sras I reconsidered the [previous discussion](https://github.com/tqtezos/baseDAO/pull/265#discussion_r639372733) and I too think we should count level discretely and not as we did with seconds/time.

So I suggest we change this `>` to `>=` here.

In fact I think there's another thing to correct, which is that in `validate_proposal_flush_expired_level` we currently enforce that the `proposal_flush_level` is has to be `> period * 2`, but I think that we should change it to be `>=
period * 2`.

Here's why: when we were dealing with time we introduced that `proposal_flush_time` needed to be more than twice the period length so that the `voting` stage of a proposal had to be ended before it could be flushed.
In other words that limitation made sure that even if a proposal was `propose`d at he very beginning of a `proposing` stage, all that stage and the next one (the `voting`) had to pass before `flush` was possible.

So let's take the implementation after this PR and the most extreme case, in which the `period` has the length of `1` level (it works for bigger periods too tho), what we would like is that it should be possible to configure the contract so that a proposal can get flushed in the first level **after** its voting period ended,
However that's not possible, `validate_proposal_flush_expired_level` now imposes that `proposal_flush_level` cannot be `<=` then `period * 2`, so in our case it needs to be at least `3`.
Moreover, this `is_level_reached` requires the number of level to fully have passed (with `>`).

So, say that with a `period` of `1` I submit a proposal in stage `x`, which is proposing, then:
- I cannot flush it in `x + 1`, which is right because this is its voting period, so this should not be allowed
- I cannot flush it in `x + 2`, which should be possible to configure, but it isn't for `validate_proposal_flush_expired_level`
- I cannot flush it in `x + 3`, which should also be possible to configure, but it isn't for `is_level_reached`

If we make the two changes instead this would be fixed, so here I suggest:
```suggestion
let is_level_reached (proposal, target : proposal * blocks): bool =
  Tezos.level >= proposal.start_level.blocks + target.blocks
```
and ... (see next comment)